components/formats-common/src/loci/common/DebugTools.java

```diff
@@ -89,24 +88,43 @@ public static synchronized boolean isEnabled() {
   }
 
   /**
-   * Attempts to enable SLF4J logging via logback or log4j
-   * without an external configuration file.
+   * Sets the root logger level.
+   *
+   * This method will override the root logger level independently of the way
+   * the logging framework has been enabled.
    *
    * @param level A string indicating the desired level
-   *   (i.e.: ALL, DEBUG, ERROR, FATAL, INFO, OFF, TRACE, WARN).
-   * @return {@code} true if logging was successfully enabled
    */
-  public static synchronized boolean enableLogging(String level) {
-    final String[][] toolClasses = new String[][] {
-      new String[] {"loci.common.", "LogbackTools"},
-      new String[] {"loci.common.", "Log4jTools"}
-    };
-
-    for (String[] toolClass : toolClasses) {
+  public static synchronized void setRootLevel(String level) {
+    for (String[] toolClass : TOOLCLASSES) {
       try {
         Class<?> k = Class.forName(toolClass[0] + toolClass[1]);
-        Method m = k.getMethod("enableLogging", String.class);
+        Method m = k.getMethod("setRootLevel", String.class);
         m.invoke(null, level);
+        return;
+      }
+      catch (Throwable t) {
```

Comment:
Catching `Throwable` is pretty dangerous, as it swallows up everything. It could hide any type of error. We should catch `ClassNotFoundException` / `NoSuchMethodException` and let other errors slip through undisturbed.